### PR TITLE
Closes #1165: Added Doorhanger to the toolbar

### LINKED ISSUE #1165: FNX-4842 ⁃ Create UI for Doorhanger be shown when the toolbar padlock is clicked
image

### PR DESCRIPTION
**Pull request by Amejia481**
[Video demo](https://drive.google.com/file/d/1PxKbabra7fiHxTMZhGxMXvG8hKna9cvQ/view?usp=sharing) 🥞 
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/QuickSettingsSheetDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+import org.jetbrains.anko.textColorResource
+import org.mozilla.fenix.utils.ItsNotBrokenSnack
+
+private const val KEY_URL = "KEY_URL"
+private const val KEY_IS_SECURED = "KEY_IS_SECURED"
+private const val KEY_IS_SITE_IN_EXCEPTION_LIST = "KEY_IS_SITE_IN_EXCEPTION_LIST"
+private const val REQUEST_CODE_QUICK_SETTINGS_PERMISSIONS = 4
+
+@SuppressWarnings("TooManyFunctions")
+class QuickSettingsSheetDialogFragment : BottomSheetDialogFragment() {
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/QuickSettingsSheetDialogFragment.kt]: There's a good bit of complexity in this Fragment and it takes some effort to read. Hence the "TooManyFunctions" lint warning. We'd normally break out the View into a UIComponent and UIView in non-trivial cases like this, leaving the Fragment only managing setup, lifecycle, and how it responds to events from the View. Do you think you could break this class out?

File: app/src/main/res/layout/fragment_quick_settings_dialog_sheet.xml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
+        android:id="@+id/quick_action_sheet"
+        android:orientation="vertical"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:contentDescription="@string/quick_settings_sheet"
+        android:background="?attr/toolbarColor">
+
+    <androidx.appcompat.widget.AppCompatTextView
```
[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/main/res/layout/fragment_quick_settings_dialog_sheet.xml]: 👍

File: app/src/main/java/org/mozilla/fenix/settings/QuickSettingsSheetDialogFragment.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/QuickSettingsSheetDialogFragment.kt]: Sure no problem!

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/QuickSettingsSheetDialogFragment.kt]: Updated!

**Comment by Amejia481** [comment]
@colintheshots I updated the PR with your suggestions

**Comment by Amejia481** [comment]
🎉

**Pull request merged**